xs/src/slic3r/Utils/Http.cpp

```diff
@@ -421,6 +421,18 @@ bool Http::ca_file_supported()
 	return res;
 }
 
+std::string Http::url_encode(const std::string &str)
+{
+	::CURL *curl = ::curl_easy_init();
+	char *ce = ::curl_easy_escape(curl, str.c_str(), str.length());
+	std::string encoded = std::string(ce);
+
+	::curl_free(ce);
+	if (curl != nullptr) { ::curl_easy_cleanup(curl); }
```

Comment:
This check should probably be performed right after `curl_easy_init()` ... Could you please move it up there? In case the curl instance is `nullptr`, let's just return an empty string. It's not ideal but I think it'll do in this case.

Reply:
I'll move it up but i think it should be moved right after ::curl_easy_escape, according the documentation
> This function must be the last function to call for an easy session

Reply:
I meant just the `nullptr` check ... that is, check for `nullptr` after creating the session, return empty string if it's `nullptr`, and at the end you can just call `::curl_easy_cleanup()` without a check...

Reply:
I've pushed a new commit according to your remarks.
I'm now returning the original string in case the curl instance is a `nullptr`. The reason is that the user is already getting a response containing an error text if the filename contains some special characters.
If the original string does not contain any special characters the upload will work, altough the curl instance is `nullptr`